src/gsi/netcdfgfs_io.f90

```diff
@@ -2327,6 +2332,447 @@ subroutine write_atm_ (grd,sp_a,filename,mype_out,gfs_bundle,ibin)
 
   end subroutine write_atm_
 
+  subroutine write_atm_pert_ (grd,sp_a,filename,mype_out,gfs_bundle,ibin)
+
+!$$$  subprogram documentation block
+!                .      .    .
+! subprogram:    write_ncatm --- Gather, transform, and write out to netcdf
+!
+!   prgmmr: whitaker         org: oar                  date: 2019-10-03
+!
+! abstract: This routine gathers fields needed for the GSI analysis
+!           file from subdomains and then transforms the fields from
+!           analysis grid to model guess grid, then written to an
+!           netcdf atmospheric analysis file.
+!
+! program history log:
+!   2019-10-03  whitaker initial version
+!
+!   input argument list:
+!     filename  - file to open and write to
+!     mype_out  - mpi task to write output file
+!    gfs_bundle - bundle containing fields on subdomains
+!     ibin      - time bin
+!
+!   output argument list:
+!
+! attributes:
+!   language: f90
+!   machines: ibm RS/6000 SP; SGI Origin 2000; Compaq HP
+!
+!$$$ end documentation block
+
+! !USES:
+    use kinds, only: r_kind,i_kind
+
+    use constants, only: r1000,fv,one,zero,qcmin,r0_05,t0c
+
+    use mpimod, only: mpi_rtype
+    use mpimod, only: mpi_comm_world
+    use mpimod, only: ierror
+    use mpimod, only: mype
+
+    use guess_grids, only: ifilesig
+    use guess_grids, only: load_geop_hgt,geop_hgti,ges_geopi
+
+    use gridmod, only: ntracer
+    use gridmod, only: ncloud
+    use gridmod, only: strip,jcap_b,bk5
+
+    use general_commvars_mod, only: load_grid,fill2_ns,filluv2_ns
+    use general_specmod, only: spec_vars
+
+    use obsmod, only: iadate
+
+    use gsi_4dvar, only: ibdate,nhr_obsbin,lwrite4danl
+    use general_sub2grid_mod, only: sub2grid_info
+    use egrid2agrid_mod,only: g_egrid2agrid,g_create_egrid2agrid,egrid2agrid_parm,destroy_egrid2agrid
+    use constants, only: two,pi,half,deg2rad
+    use gsi_bundlemod, only: gsi_bundle
+    use gsi_bundlemod, only: gsi_bundlegetpointer
+    use cloud_efr_mod, only: cloud_calc_gfs
+
+    use netcdf, only: nf90_max_name
+    use module_ncio, only: open_dataset, close_dataset, Dimension, Dataset,&
+         read_attribute, write_attribute,get_dim, create_dataset, write_vardata, read_vardata,&
+         get_idate_from_time_units,quantize_data,get_time_units_from_idate,has_attr,has_var
+    use ncepnems_io, only: error_msg
+
+    implicit none
+
+! !INPUT PARAMETERS:
+
+    type(sub2grid_info), intent(in) :: grd
+    type(spec_vars),     intent(in) :: sp_a
+    character(len=24),   intent(in) :: filename  ! file to open and write to
```

Comment:
Increase to at least `len=70` to be consistent with size of `filename` in `cplr_gfs_ensmod.f90`.  Is there a way to move away from a fixed character length?   It's possible `len=70` could fail.